src/scores.ts

```diff
@@ -108,8 +109,21 @@ export async function updateProposalAndVotes(proposalId: string, force = false)
     return true;
   }
 
-  // Ignore score calculation if proposal have more than 100k votes and scores_updated greater than 5 minute
   const ts = Number((Date.now() / 1e3).toFixed());
+
+  // Delay computation of final scores, to allow time for last minute votes to finish
+  // up to 1 minute after the end of the proposal
+  if (proposal.end <= ts && proposal.scores_state !== 'final') {
+    const secondsSinceEnd = ts - proposal.end;
+    await snapshot.utils.sleep(
+      Math.min(
+        Math.max(FINALIZE_SCORE_SECONDS_DELAY - secondsSinceEnd, 0),
+        FINALIZE_SCORE_SECONDS_DELAY
+      ) * 1000
```

Comment:
`Math.max(FINALIZE_SCORE_SECONDS_DELAY - secondsSinceEnd, 0)` can never be bigger than `FINALIZE_SCORE_SECONDS_DELAY` right? 🤔 we can remove `Math.min` here and use `Math.max(FINALIZE_SCORE_SECONDS_DELAY - secondsSinceEnd, 0)`

Reply:
`FINALIZE_SCORE_SECONDS_DELAY - secondsSinceEnd` can be negative, in the event the proposal has closed for more than 60 seconds

Reply:
No i am talking about `Math.min` not `max`, max is fine 😅 

`Math.max(FINALIZE_SCORE_SECONDS_DELAY - secondsSinceEnd, 0)` will always be `<= FINALIZE_SCORE_SECONDS_DELAY` (since we are removing something from it) and `>= 0`, so no benefit of using `Math.min` here, It will never pick 
```suggestion
      Math.max(FINALIZE_SCORE_SECONDS_DELAY - secondsSinceEnd, 0) * 1000
```

Reply:
You’re right